packages/modals/src/elements/Modal.spec.tsx

```diff
@@ -9,27 +9,27 @@ import React from 'react';
 import { render, fireEvent } from 'garden-test-utils';
 import { KEY_CODES } from '@zendeskgarden/container-utilities';
 
-import Modal from './Modal';
-import Body from '../views/Body';
-import Footer from '../views/Footer';
-import Header from '../views/Header';
-import Close from '../views/Close';
+import { Modal, IModalProps } from './Modal';
+import { Body } from './Body';
+import { Footer } from './Footer';
+import { Header } from './Header';
+import { Close } from './Close';
 
 describe('Modal', () => {
   const MODAL_ID = 'TEST_ID';
-  let onCloseSpy;
+  let onCloseSpy: jest.Mock;
 
-  const BasicExample = props => (
+  const BasicExample = (props: IModalProps) => (
     <Modal
       {...props}
       id={MODAL_ID}
       data-test-id="modal"
-      backdropProps={{ 'data-test-id': 'backdrop' }}
+      backdropProps={{ 'data-test-id': 'backdrop' } as any}
```

Comment:
I'm surprised `data-*` attributes aren't included with HTMLAttributes used in the props interface. If this is needed we may want to just type `backdropProps` as `any`.

Reply:
I was surprised too!

One concern for typing `backdropProps` as `any` is that it might result in the prop sheet communicating to consumers that `any` type can be passed for `backdropProps`, even strings, booleans, etc. By casting it here, consumers see the prop sheet type as:

```ts
HTMLAttributes<HTMLDivElement> | undefined
```

This way they are informed that only `div` attributes can be passed to `backdropProps`. Is this a valid concern? Let me know what you think.

Reply:
This should be fine! It's more specific to the actual use-case as well.